src/components/create-safe/steps/OwnerRow.tsx

```diff
@@ -36,30 +34,23 @@ export const OwnerRow = ({
     [getValues],
   )
 
-  const { name: fallbackName, resolving } = useAddressResolver(owner.address)
+  const { ens, resolving } = useAddressResolver(owner.address)
 
   useEffect(() => {
-    if (!owner.name && fallbackName) {
-      setValue(`owners.${index}.fallbackName`, fallbackName)
+    if (ens) {
+      setValue(`owners.${index}.ens`, ens)
     }
-  }, [fallbackName, setValue, owner.name, index])
+  }, [ens, setValue, index])
 
   return (
-    <Grid
-      container
-      key={field.id}
-      spacing={3}
-      alignItems="center"
-      marginBottom={3}
-      flexWrap={['wrap', undefined, 'nowrap']}
-    >
+    <Grid container spacing={3} alignItems="center" marginBottom={3} flexWrap={['wrap', undefined, 'nowrap']}>
       <Grid item xs={12} md={4}>
         <FormControl fullWidth>
           <NameInput
             name={`owners.${index}.name`}
             label="Owner name"
             InputLabelProps={{ shrink: true }}
-            placeholder={fallbackName}
+            placeholder={ens || `Owner ${index + 1}`}
```

Comment:
Why not a mnemonic here?

Reply:
As we don't set the fallback as the `name` anymore, the mnemonic will be different when going forward/back. Should I add it back?

Reply:
If the mnemonic is not being used anymore, should we still display it?

Reply:
As long as it stays the same while staring at the same component, it's fine.

Reply:
It's being used for Safe name and when adding a new owner, but perhaps when adding a new owner it would make sense to align with here and label it "New owner".

Reply:
Actually yeah, the whole point of the mnemonic was to avoid empty names in the address book. If we're not saving the mnemonic, we could as well not show anything...
I like the quirky animal names tho.

Reply:
Probably we should change the mnemonic generation in a way that the generated mnemonic is deterministic for each address? So it always generates the same name for each address.

Reply:
I like the idea @schmanu but think it's something to tackle in a separate PR.